client/core/trade.go

```diff
@@ -2214,6 +2238,105 @@ func (c *Core) sendRedeemAsync(t *trackedTrade, match *matchTracker, coinID, sec
 	}()
 }
 
+// redemptionFeeSuggesiton gets a suggestion for a redemption transaction.
+// It avoids using (*Core).feeSuggestion, since it can incur an RPC request.
+// t.redeemFeeSuggestion is updated on every tck and uses a rate directly
+// from our wallet, if available. Only go looking for a rate if we don't
+// have one cached.
+func (t *trackedTrade) redemptionFeeSuggestion() uint64 {
+	var feeSuggestion uint64
+	if _, is := t.accountRedeemer(); is {
+		feeSuggestion = t.metaData.RedeemMaxFeeRate
+	} else {
+		feeSuggestion = atomic.LoadUint64(&t.redeemFeeSuggestion)
+	}
+	if feeSuggestion == 0 {
+		feeSuggestion = t.dc.bestBookFeeSuggestion(t.wallets.toAsset.ID)
+	}
+
+	return feeSuggestion
+}
+
+// resubmitRedemption resubmits the redemption transaction for a match. It also
+// updates the match metadata regarding the redemption.
+func (t *trackedTrade) resubmitRedemption(match *matchTracker) error {
+	coinID, _, fees, err := t.wallets.toWallet.Redeem(&asset.RedeemForm{
+		Redemptions: []*asset.Redemption{
+			{
+				Spends: match.counterSwap,
+				Secret: match.MetaData.Proof.Secret,
+			},
+		},
+		FeeSuggestion: t.redemptionFeeSuggestion(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to resubmit redemption: %v", err)
+	}
+
+	// Should always be 1 unless Redeem has a bug, but still updating the
+	// match metadata with the first redeem if there are more than 1 in
+	// the slice.
+	if len(coinID) < 1 {
+		return fmt.Errorf("no coinIDs returned from Redeem")
+	}
+
+	if match.Side == order.Maker {
+		match.MetaData.Proof.MakerRedeem = order.CoinID(coinID[0])
+	} else {
+		match.MetaData.Proof.TakerRedeem = order.CoinID(coinID[0])
+	}
+
+	// TODO: need to get rid of previously paid fees, or just not add these
+	// new ones? This field is only for informational purposes.
+	t.metaData.RedemptionFeesPaid += fees
```

Comment:
I think it's kinda messy to try to replace. Probably not so different than the first try. Just ignore imo...

Reply:
I tend to agree.